### PR TITLE
use parameter containerized for NewMainKubelet

### PR DESCRIPTION
**Pull request by wackxu**
use parameter containerized for NewMainKubelet

```release-note
NONE
```

### CODE REVIEW

File: pkg/kubelet/kubelet.go [hunk 1]
```diff
@@ -548,6 +548,7 @@ func NewMainKubelet(kubeCfg *kubeletconfiginternal.KubeletConfiguration,
 		iptablesDropBit:                         int(kubeCfg.IPTablesDropBit),
 		experimentalHostUserNamespaceDefaulting: utilfeature.DefaultFeatureGate.Enabled(features.ExperimentalHostUserNamespaceDefaultingGate),
 		keepTerminatedPodVolumes:                keepTerminatedPodVolumes,
+		containerized:                           containerized,
```
Comment (dixudx) [inline pkg/kubelet/kubelet.go]: Actually I would like to remove param `containerized` from `NewMainKubelet`. This param is redundant.

`s.Containerized` is already utilized in `cmd/kubelet/app/server.go`. 

/cc @mtaufen This api change is introduced by your #53088.

**Review by dixudx — CHANGES_REQUESTED** [review]

File: pkg/kubelet/kubelet.go (hunk 1 above)
Comment (wackxu) [reply pkg/kubelet/kubelet.go]: In `cmd/kubelet/app/server.go`, the sequence of function call is builder --> CreateAndInitKubelet --> NewMainKubelet, so the `containerized` is not utilized

Comment (dixudx) [reply pkg/kubelet/kubelet.go]: I mean `containerized` is redundant in `NewMainKubelet`.

https://github.com/kubernetes/kubernetes/blob/160270800f099538d9ba888201737b4dbab7587a/cmd/kubelet/app/server.go#L141-L145

We won't need  `containerized` for `NewMainKubelet` any more.

Comment (wackxu) [reply pkg/kubelet/kubelet.go]: @dixudx Thanks for your clarify, then I think we can remove the parameter and the field for `kubelet` struct

**Comment by mtaufen** [comment]
@wackxu let me know when you have a PR to remove the unnecessary field :)

**Comment by dixudx** [comment]
close this pr. I prefer #56757. @wackxu You can reopen it if you need.

/close

**Pull request closed without merging**